thunder/dynamo/compiler.py

```diff
@@ -41,18 +43,40 @@
 _DEFAULT_THUNDERFX_DISABLE_SPLIT_AUTOGRAD = True
 
 
-def _add_prologue_pruning(options: dict):
-    """
-    Add a transform to prune prologue checks to the list of transforms in the given options dictionary.
+def is_in_torch_compile() -> bool:
+    """Returns :obj:`True` if :func:`torch.compile` is active."""
+    return TorchCompileContext.current_compile_id() is not None
 
-    Args:
-        options: The dictionary of options to modify
+
+def is_dynamic_inputs(example_inputs):
+    """Check if inputs dynamic or not by checking the presence of :class:`torch.SymInt`."""
+    from torch.utils import _pytree as pytree
+
+    flat_example_inputs, _ = pytree.tree_flatten(example_inputs)
+    return any(
+        isinstance(a, torch.SymInt) or any(isinstance(s, torch.SymInt) for s in a.shape) for a in flat_example_inputs
```

Comment:
I think it might be possible that `a` is not a torch.SymInt or a torch.Tensor in which case this would error at `a.shape`.

`a` could be a tuple of `Tensor` or such as well.

Reply:
I think it could be any type other than `torch.SymInt` or `torch.Tensor` but I think thanks to `tree_flatten`, containers are not there